### PR TITLE
Oliinyk / 2152 Create workshop email validation

### PR DESCRIPTION
**Pull request by doliinyk**
#2152 

Add email regex and email validator, replaced default Angular email validators with created validator.
Refactored regex patterns and validators.

### CODE REVIEW

**Comment by doliinyk** [comment]
TODO: Write hint for new validator for `app-validation-hint` (now error hint is empty, just red margin of input there)

**Review by humanmantis — APPROVED** [review]

File: src/assets/i18n/uk.json [hunk 1]
```diff
@@ -396,7 +396,7 @@
     "UAH_FOR": "грн за",
     "VALIDATIONS": {
       "INVALID_BIRTHDAY": "Дата народження виходить за межі допустимих.",
-      "INVALID_CHARACHTERS": ", використовуйте будь-ласка тільки кирилицю",
+      "INVALID_CHARACTERS": ", використовуйте будь-ласка тільки кирилицю",
```
Comment (humanmantis) [inline src/assets/i18n/uk.json]: This is unrelated to code, but grammatically more correct version is: `, використовуйте, будь ласка, тільки кирилицю`

File: src/app/shared/components/confirmation-modal-window/reason-modal-window/reason-modal-window.component.html [hunk 2]
```diff
@@ -20,8 +20,8 @@ <h2 mat-dialog-title>{{ modalTitle | translate }}</h2>
 
     <app-validation-hint
       [validationFormControl]="ReasonFormControl"
-      [minCharachters]="validationConstants.INPUT_LENGTH_1"
-      [maxCharachters]="validationConstants.MAX_DESCRIPTION_LENGTH_500"
+      [minCharacters]="validationConstants.INPUT_LENGTH_1"
+      [maxCharacters]="validationConstants.MAX_DESCRIPTION_LENGTH_500"
     >
```
Comment (humanmantis) [inline src/app/shared/components/confirmation-modal-window/reason-modal-window/reason-modal-window.component.html]: Keep on same line

File: src/app/shell/personal-cabinet/parent/create-child/child-form/child-form.component.html [hunk 3]
```diff
@@ -11,8 +11,8 @@
   </mat-form-field>
   <app-validation-hint
     [validationFormControl]="ChildFormGroup.get('lastName')"
-    [minCharachters]="validationConstants.INPUT_LENGTH_1"
-    [maxCharachters]="validationConstants.INPUT_LENGTH_60"
+    [minCharacters]="validationConstants.INPUT_LENGTH_1"
+    [maxCharacters]="validationConstants.INPUT_LENGTH_60"
   >
```
Comment (humanmantis) [inline src/app/shell/personal-cabinet/parent/create-child/child-form/child-form.component.html]: same

**Review by humanmantis — CHANGES_REQUESTED** [review]

**Review by humanmantis — APPROVED** [review]

**Pull request merged**